tracing.go

```diff
@@ -22,17 +22,31 @@ type tracingInfoKey struct{}
 
 // NewOpenTracingHooks provides a twirp.ServerHooks struct which records
 // OpenTracing spans.
-func NewOpenTracingHooks(tracer ot.Tracer) *twirp.ServerHooks {
+func NewOpenTracingHooks(tracer ot.Tracer, opts ...HookOpt) *twirp.ServerHooks {
 	hooks := &twirp.ServerHooks{
 		RequestReceived: startTraceSpan(tracer),
 		RequestRouted:   handleRequestRouted,
 		ResponseSent:    finishTrace,
-		Error:           handleError,
+		Error:           handleError(true),
+	}
+
+	for _, opt := range opts {
+		opt(hooks)
 	}
 
 	return hooks
 }
 
+type HookOpt func(h *twirp.ServerHooks)
+
+// WithUserErr, if set, will report client errors (4xx) as errors in the server span.
+// If not set, only 5xx status will be reported as erroneous.
+func WithUserErr(withUserError bool) HookOpt {
```

Comment:
Similar to the client, how do you feel about changing this to be `IncludeClientErrors` instead?

Reply:
Done!  Tho, one thing I'm not the biggest fan of is that since these are in the same package they share a namespace, so both options can't be called `IncludeClientErrors`.  I ended up making the client one `IncludeUserErrors`, since from its perspective it's making the "error" but it's a user-class code, and the server one `IncludeClientErrors`, since its client-side for the server, but I don't think that's perfect/is still confusing.

If you have better recommendations on naming LMK.

Reply:
Thanks for changing this up! So, looking at the [grpc implementation](https://github.com/grpc-ecosystem/go-grpc-middleware/blob/master/tracing/opentracing/client_interceptors.go) (which is what inspired this package), they actually share the same option. Should we consolidate these options on our end as well? 🤔 Maybe into `TraceOption`?

Reply:
Done and done!